sanity/doctors/ApprovalDoctor.js

```diff
@@ -0,0 +1,49 @@
+import { Doctor } from './Doctor'
+import { Diagnosis } from './Diagnosis'
+import { BlockchainEvent } from '../../src/BlockchainEvent'
+import { eventNames } from '../../src/ethereum'
+
+export class ApprovalDoctor extends Doctor {
+  async diagnose() {
+    return new ApprovalDiagnosis()
+  }
+}
+
+export class ApprovalDiagnosis extends Diagnosis {
+  constructor() {
+    super()
+    this.faultyAssets = []
+    this.fromBlock = 0
+  }
+
+  async getFaultyAssets() {
+    return this.faultyAssets
+  }
+
+  hasProblems() {
+    // Always run this doctor.
+    // There is no way to check if a specific approval is missing because of how the contract
+    // stores those mappings
+    return true
+  }
+
+  async prepare(fromBlock) {
+    this.fromBlock = fromBlock || 0
```

Comment:
Small comment. Because we're using js and we don't have type safety, we might want to do this:

`Number(fromBlock) || 0`

just in case